File: MsgServerClientSdk/SdkMsgClientAndroid/sdkmsgclient/src/main/java/org/dync/teameeting/sdkmsgclient/msgs/MSGroupManager.java
package org.dync.teameeting.sdkmsgclient.msgs;

/**
 * Created by hp on 7/10/16.
 */
public class MSGroupManager {
    /**
     *  add group delegate
     *
     *  params:
     *      grpDelegate: the group delegate for callback to set
     *
     *  return:
     *      void
     */
    public void addDelegate(MSGroupDelegate grpDelegate) {
        MsgClient.getInstance().MCSetGroupDelegate(grpDelegate);
    }

    /**
     *  remove group delegate
     *
     *  params:
     *      grpDelegate: the group delegate to delete
     *
     *  return:
     *      void
     */
    public void delDelegate(MSGroupDelegate grpDelegate) {
        MsgClient.getInstance().MCSetGroupDelegate(null);
    }

    /**
     *  add to group
     *
     *  params:
     *      grpId: the group id to add
     *
     *  return:
     *      void
     */
    public void addGroup(String grpId) {
        if (grpId.length()==0) return;
        MsgClient.getInstance().MCAddGroup(grpId);
    }

    /**
     *  remove from group
     *
     *  params:
     *      grpId: the group id to remove
     *
     *  return:
     *      void
     */
    public void rmvGroup(String grpId) {
        if (grpId.length()==0) return;
        MsgClient.getInstance().MCRmvGroup(grpId);
    }
}

File: MsgServer/MsgServerClient/ClientCommon/MSGroupManager.cpp
//
//  MSGroupManager.m
//  SdkMsgClientIos
//
//  Created by hp on 6/29/16.
//  Copyright © 2016 Dync. All rights reserved.
//

#include "MSGroupManager.h"
#include "MsgClient.h"

void MSGroupManager::AddDelegate(const MSGroupDelegate* grpDelegate)
{
    MsgClient::Instance().MCSetGroupDelegate(grpDelegate);
}

void MSGroupManager::DelDelegate(const MSGroupDelegate* grpDelegate)
{
    MsgClient::Instance().MCSetGroupDelegate(nullptr);
}

void MSGroupManager::AddGroup(const std::string& grpId)
{
    if (grpId.length()==0) return;
    MsgClient::Instance().MCAddGroup(grpId);
}

void MSGroupManager::RmvGroup(const std::string& grpId)
{
    if (grpId.length()==0) return;
    MsgClient::Instance().MCRmvGroup(grpId);
}
